https://github.com/sihsieh65-tech/cs82a-portfolio/upload/main/Module3

Data from the usarrests.csv file were loaded. I ran the .shape, .info(), .describe(), and .isna().sum() functions. 

In [1]:
import pandas as pd
df = pd.read_csv('usarrests.csv')
df

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91
5,Colorado,7.9,204.0,78
6,Connecticut,3.3,110.0,77
7,Delaware,5.9,238.0,72
8,Florida,15.4,335.0,80
9,Georgia,17.4,NaN,60


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  50 non-null     str    
 1   Murder      50 non-null     float64
 2   Assault     49 non-null     float64
 3   UrbanPop    50 non-null     int64  
dtypes: float64(2), int64(1), str(1)
memory usage: 1.7 KB


In [3]:
df.shape

(50, 4)

In [4]:
df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [5]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       1
UrbanPop      0
dtype: int64

Two values in the table strike me as suspect. Firstly, Iowa shows 570% of the population lives in an urban area, a value that is impossible as you cant have more 5.7 times the state population living in an urban area. The second suspect datapoint is the 6% urban population value for New York. Using google i was able to locate a report from the 1970 census which shows the percent urban population of New York to be 85.6% (pdf for the report will be submitted). I also checked the other UrbanPop numbers and Iowas number was actually 57.2%. Interestigly the number in the dataset for Maryland was reported as 67% but this report said it should be 76.6%. I am inclined to believe the report since 47 of 50 states show agreement with the numbers and the dataset is already known to contain errors. I believe the numbers should be corrected to match the 1970 census data.


In [6]:
df["UrbanPop"]=df["UrbanPop"].replace(570,57)
df["UrbanPop"]=df["UrbanPop"].replace(6,86)

In [7]:
df

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91
5,Colorado,7.9,204.0,78
6,Connecticut,3.3,110.0,77
7,Delaware,5.9,238.0,72
8,Florida,15.4,335.0,80
9,Georgia,17.4,NaN,60


Correcting the incorrect Maryland data.

In [8]:
df.loc[19, 'UrbanPop'] = 77

In [9]:
df

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91
5,Colorado,7.9,204.0,78
6,Connecticut,3.3,110.0,77
7,Delaware,5.9,238.0,72
8,Florida,15.4,335.0,80
9,Georgia,17.4,NaN,60


Examining the assualt data the South Carolina value of 879 assaults per 100000 people seems like a huge outlier. No other state reports a similarly large number, the second closest is North Carolina at 337. While it does seem extreme it does not appear to be impossible on its face.  